### PR TITLE
feat: Implement core LLM output extraction logic (Task 1)

### PR DESCRIPTION
**Pull request by nagyv**
## Summary
- Implemented core LLM output extraction functions for reading transcript files and parsing assistant messages
- Added comprehensive unit tests with 89% code coverage
- Completed all Task 1 requirements from issue #13

## Changes
- ✅ Implemented `parse_hook_input()` - Parses and validates Stop hook JSON input from stdin
- ✅ Implemented `read_transcript_file()` - Reads JSONL transcript files and extracts last assistant message
- ✅ Implemented `extract_llm_metadata()` - Extracts text content and usage statistics from assistant messages
- ✅ Created comprehensive unit tests covering all edge cases
- ✅ Achieved 89% test coverage (missing lines are for unimplemented Task 2/3 functions)

## Test Plan
- [x] Unit tests pass with 89% coverage
- [x] Handles valid transcript files with full metadata
- [x] Handles minimal data scenarios
- [x] Gracefully handles malformed JSON
- [x] Proper error handling for missing files
- [x] Tests verify all BDD scenarios for extraction logic

## Next Steps
- Task 2: Implement PostgreSQL database integration
- Task 3: Wire components together into working Stop hook
- Task 4: Add BDD tests and documentation

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: tests/claude/scripts/test_save_llm_output.py [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+        
+        result = save_llm_output.extract_llm_metadata(assistant_message)
+        
+        assert result['output'] == "Response"
+        assert result['input_tokens'] is None
+        assert result['output_tokens'] is None
+        assert result['service_tier'] is None
+
+
+class TestIntegration:
```
Comment (nagyv) [inline tests/claude/scripts/test_save_llm_output.py:350]: Drop this. Another task will add integration tests.
